echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/microsoftteams/MicrosoftTeamsService.java

```diff
@@ -16,18 +16,64 @@
 
 package com.netflix.spinnaker.echo.microsoftteams;
 
+import com.netflix.spinnaker.retrofit.Slf4jRetrofitLogger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import lombok.extern.slf4j.Slf4j;
+import retrofit.RestAdapter;
+import retrofit.client.Client;
 import retrofit.client.Response;
+import retrofit.converter.JacksonConverter;
 
 @Slf4j
 public class MicrosoftTeamsService {
-  private final MicrosoftTeamsClient microsoftTeamsClient;
+  private static final String BASE_URL_REGEX_PATTERN = "^(http|https)://.+?/";
 
-  public MicrosoftTeamsService(MicrosoftTeamsClient microsoftTeamsClient) {
-    this.microsoftTeamsClient = microsoftTeamsClient;
+  private Client retrofitClient;
+  private RestAdapter.LogLevel retrofitLogLevel;
+
+  public MicrosoftTeamsService(Client retrofitClient, RestAdapter.LogLevel retrofitLogLevel) {
+    this.retrofitClient = retrofitClient;
+    this.retrofitLogLevel = retrofitLogLevel;
   }
 
   public Response sendMessage(String webhookUrl, MicrosoftTeamsMessage message) {
-    return microsoftTeamsClient.sendMessage(webhookUrl, message);
+    MicrosoftTeamsClient microsoftTeamsClient =
```

Comment:
do we _have_ to create a new client per message now? Is that because the domain may be something different from outlook.office.com?

Reply:
Unfortunately, yes. The domain could be something like outlook.office.com or [tenant-name].webhook.office.com, and the webhook path is slightly different now, too. This seemed to me to be the best option since the URL could be different even at an individual Teams channel level.

Reply:
I see, thanks! I'd be really happy if you could add this explanation as a comment. Otherwise LGTM 👍

Reply:
No problem - I added a comment here.